Common/src/main/java/com/tabnineCommon/chat/ChatFrame.kt

```diff
@@ -27,13 +28,25 @@ import javax.swing.event.HyperlinkEvent
 class ChatFrame(private val project: Project, private val binaryRequestFacade: BinaryRequestFacade) :
     JPanel(true), Disposable {
     private var capabilitiesFetched = false
+    private var isLoggedIn = false
 
     init {
         layout = BorderLayout()
 
         updateDisplay()
 
         val connection = ApplicationManager.getApplication().messageBus.connect(this)
+
+        connection.subscribe(
+            BinaryStateChangeNotifier.STATE_CHANGED_TOPIC,
+            BinaryStateChangeNotifier { state ->
+                isLoggedIn = state.isLoggedIn == true
+                ApplicationManager.getApplication().invokeLater {
```

Comment:
we're having serious issues with the locks on the UI thread. 
is this a safe way to do it

Reply:
yes!